app/plugin.go

```diff
@@ -255,7 +256,20 @@ func (a *App) SyncPlugins() *model.AppError {
 			continue
 		}
 		defer reader.Close()
-
+		var sigReader filesstore.ReadCloseSeeker
+		sigReader, appErr = a.FileReader(fmt.Sprintf("%s.sig", path))
```

Comment:
Should we invert these checks to only look for and read a signature if `RequirePluginSignature`?

Reply:
We discussed this with @ali-farooq0, our understanding is that we should still verify if there is a signature file, but if there is non and `RequirePluginSignature` is set throw an error and not install plugin on synchronization.

Reply:
Discussed offline, but my thinking is still that there's no point in checking a signature if we're not going to enforce it when it's absent. If we still try to force validation when the feature is disabled, it will be hard to remove a trusted certificate without breaking all the plugins.